src/scalper.ts

```diff
@@ -274,15 +328,54 @@ export class Scalper {
     );
 
     if ((netGamma * fairValue) < (this.tickSize * fairValue)){
-      console.log(this.symbol, 'Gamma Hedge too small')
+      console.log(this.symbol, 'Gamma Hedge Too Small')
       return
     }
 
+    const gammaFillListener = (event) => {
+      const parsedEvent = JSON.parse(event.data);
+      if (
+        parsedEvent['status'] === 'New' &&
+        parsedEvent['market'] === this.symbol.concat("-PERP")
+      ) {
+        const fillBytes = Buffer.from(parsedEvent['event'], 'base64');
+        const fillEvent: FillEvent = PerpEventLayout.decode(fillBytes).fill;
+        if (
+          (fillEvent.makerClientOrderId.toString() == gammaBidID.toString()) ||
+          (fillEvent.takerClientOrderId.toString() == gammaBidID.toString())
+        ) {
+          console.log(this.symbol, 'Gamma Bid Filled', gammaBidID, new Date().toUTCString());
+          fillFeed.removeEventListener('message', gammaFillListener);
+          this.gammaScalp(
```

Comment:
Add a todo to try to remove recursion or add explicit termination, we dont want recursion in production code since it leads to difficult to debug problems when possible.